vllm/model_executor/models/llava_next.py

```diff
@@ -255,40 +256,20 @@ def _validate_pixel_values(
         self, data: Union[torch.Tensor, List[torch.Tensor]]
     ) -> Union[torch.Tensor, List[torch.Tensor]]:
 
+        h = w = self.config.vision_config.image_size
+        expected_dims = (3, h, w)
+
         def _validate_shape(data: torch.Tensor):
+            actual_dims = tuple(data.shape[1:])
 
-            dim = data.dim()
-            height = width = self.config.vision_config.image_size
-            # All 4d image tensors have the same number of patches,
-            # so data is a 5d batch of these tensors
-            if dim == 5:
-                if list(data.shape)[2:] != [
-                        3, self.config.vision_config.image_size,
-                        self.config.vision_config.image_size
-                ]:
-                    raise ValueError(
-                        "Expected pixel value tensor in shape of: (batch size, "
-                        f"patch number, 3, {height}, {width}), got {data.shape}"
-                    )
-
-            # 4d image tensors have different number of patches,
-            # so data is each individual tensor.
-            elif dim == 4:
-                if list(data.shape)[1:] != [
-                        3, self.config.vision_config.image_size,
-                        self.config.vision_config.image_size
-                ]:
-                    raise ValueError(
-                        "Expected pixel value tensor in shape of: (patch "
-                        f"number, 3, {height}, {width}), got {data.shape}")
-            else:
+            if actual_dims != expected_dims:
                 raise ValueError(
-                    f"Invalid pixel value tensor of shape {data.shape}")
+                    "The expected shape of pixel values in each batch element "
+                    f"is number of patches plus {actual_dims}. You supplied "
+                    f"{tuple(data.shape)}.")
 
-        if isinstance(data, torch.Tensor):
-            _validate_shape(data)
-        else:
-            [_validate_shape(d) for d in data]
+        for d in data:
+            _validate_shape(d)
```

Comment:
I don't think this will work? data is either a 5d tensor or a list of 4d tensor, correct?

Reply:
Yes, but in either case, iterating through it will yield 4D tensors which simplifies the logic somewhat.

Reply:
Ah my bad - I was thinking feeding `data` into  `_validate_shape` directly. Yes this works!